azure: ci e2e timing out

sadly the test output is not exactly parseable (to me), but it looks the azure e2e tests are timing out, because they fail to come and got deleted within 120s, I think this needs to be adjusted.

```
--- FAIL: TestCreatePodWithConfigMapAzure (111.76s)
    --- FAIL: TestCreatePodWithConfigMapAzure/ConfigMapPeerPod_test (111.76s)
        --- PASS: TestCreatePodWithConfigMapAzure/ConfigMapPeerPod_test/Configmap_is_created_and_contains_data (1.13s)
=== NAME  TestCreatePodWithSecretAzure/SecretPeerPod_test/Secret_has_been_created_and_contains_data
    assessment_runner.go:409: Output when execute test commands:
time="2024-02-20T14:08:06Z" level=info msg="VM found in resource group"
time="2024-02-20T14:08:06Z" level=info msg="Deleting Secret... busybox-secret"
time="2024-02-20T14:08:06Z" level=info msg="Deleting pod busybox-secret-pod..."
=== NAME  TestCreateSimplePodAzure/SimplePeerPod_test
    assessment_runner.go:259: Waiting for containers in pod: simple-test are ready
=== RUN   TestCreateSimplePodAzure/SimplePeerPod_test/PodVM_is_created
time="2024-02-20T14:08:36Z" level=info msg="VM found in resource group"
time="2024-02-20T14:08:36Z" level=info msg="Deleting pod simple-test..."
time="2024-02-20T14:08:51Z" level=info msg="Pod busybox-secret-pod has been successfully deleted within 60s"
--- PASS: TestCreatePodWithSecretAzure (96.09s)
    --- PASS: TestCreatePodWithSecretAzure/SecretPeerPod_test (96.09s)
        --- PASS: TestCreatePodWithSecretAzure/SecretPeerPod_test/Secret_has_been_created_and_contains_data (0.53s)
=== NAME  TestCreateSimplePodAzure/SimplePeerPod_test
    assessment_runner.go:561: timed out waiting for the condition
--- FAIL: TestCreateSimplePodAzure (105.89s)
    --- FAIL: TestCreateSimplePodAzure/SimplePeerPod_test (105.89s)
        --- PASS: TestCreateSimplePodAzure/SimplePeerPod_test/PodVM_is_created (0.44s)
```

--- a/test/e2e/assessment_helpers.go
+++ b/test/e2e/assessment_helpers.go
@@ -60,6 +60,7 @@ func NewTestCase(t *testing.T, e env.Environment, testName string, assert CloudA
 		podState:       v1.PodRunning,
 		imagePullTimer: false,
 		isAuth:         false,
+		deletionWithin: assert.DefaultTimeout(),
 	}
 
 	return testCase

--- a/test/e2e/assessment_runner.go
+++ b/test/e2e/assessment_runner.go
@@ -75,7 +75,7 @@ type TestCase struct {
 	imagePullTimer       bool
 	isAuth               bool
 	AuthImageStatus      string
-	deletionWithin       *time.Duration
+	deletionWithin       time.Duration
 	testInstanceTypes    InstanceValidatorFunctions
 	isNydusSnapshotter   bool
 }
@@ -121,7 +121,7 @@ func (tc *TestCase) WithService(service *v1.Service) *TestCase {
 }
 
 func (tc *TestCase) WithDeleteAssertion(duration *time.Duration) *TestCase {
-	tc.deletionWithin = duration
+	tc.deletionWithin = *duration
 	return tc
 }
 
@@ -546,10 +546,6 @@ func (tc *TestCase) Run() {
 			}
 
 			if tc.pod != nil {
-				duration := 1 * time.Minute
-				if tc.deletionWithin == nil {
-					tc.deletionWithin = &duration
-				}
 				if err = client.Resources().Delete(ctx, tc.pod); err != nil {
 					t.Fatal(err)
 				}
@@ -557,7 +553,7 @@ func (tc *TestCase) Run() {
 				if err = wait.For(conditions.New(
 					client.Resources()).ResourceDeleted(tc.pod),
 					wait.WithInterval(5*time.Second),
-					wait.WithTimeout(*tc.deletionWithin)); err != nil {
+					wait.WithTimeout(tc.deletionWithin)); err != nil {
 					t.Fatal(err)
 				}
 				log.Infof("Pod %s has been successfully deleted within %.0fs", tc.pod.Name, tc.deletionWithin.Seconds())
@@ -565,7 +561,7 @@ func (tc *TestCase) Run() {
 
 			if tc.extraPods != nil {
 				for _, extraPod := range tc.extraPods {
-					err := DeletePod(ctx, client, extraPod.pod, tc.deletionWithin)
+					err := DeletePod(ctx, client, extraPod.pod, &tc.deletionWithin)
 					if err != nil {
 						t.Logf("Error occurs when delete pod: %s", extraPod.pod.Name)
 						t.Fatal(err)

--- a/test/e2e/aws_common.go
+++ b/test/e2e/aws_common.go
@@ -7,6 +7,7 @@ import (
 	"context"
 	"strings"
 	"testing"
+	"time"
 
 	"github.com/aws/aws-sdk-go-v2/aws"
 	"github.com/aws/aws-sdk-go-v2/service/ec2"
@@ -23,6 +24,10 @@ func NewAWSAssert() AWSAssert {
 	return AWSAssert{
 		Vpc: pv.AWSProps.Vpc,
 	}
+}
+
+func (aa AWSAssert) DefaultTimeout() time.Duration {
+	return 1 * time.Minute
 }
 
 func (aa AWSAssert) HasPodVM(t *testing.T, id string) {

--- a/test/e2e/azure_common.go
+++ b/test/e2e/azure_common.go
@@ -7,6 +7,7 @@ import (
 	"context"
 	"strings"
 	"testing"
+	"time"
 
 	pv "github.com/confidential-containers/cloud-api-adaptor/test/provisioner/azure"
 	log "github.com/sirupsen/logrus"
@@ -37,6 +38,10 @@ func checkVMExistence(resourceGroupName, prefixName string) bool {
 	}
 
 	return false
+}
+
+func (c AzureCloudAssert) DefaultTimeout() time.Duration {
+	return 2 * time.Minute
 }
 
 func (c AzureCloudAssert) HasPodVM(t *testing.T, id string) {

--- a/test/e2e/common.go
+++ b/test/e2e/common.go
@@ -255,6 +255,7 @@ func WaitForClusterIP(t *testing.T, client klient.Client, svc *v1.Service) strin
 type CloudAssert interface {
 	HasPodVM(t *testing.T, id string)                             // Assert there is a PodVM with `id`.
 	GetInstanceType(t *testing.T, podName string) (string, error) // Get Instance Type of PodVM
+	DefaultTimeout() time.Duration                                // Default timeout for cloud operations
 }
 
 // RollingUpdateAssert defines assertions for rolling update test

--- a/test/e2e/common_suite.go
+++ b/test/e2e/common_suite.go
@@ -40,7 +40,7 @@ func DoTestCreateSimplePodWithNydusAnnotation(t *testing.T, e env.Environment, a
 
 func DoTestDeleteSimplePod(t *testing.T, e env.Environment, assert CloudAssert) {
 	pod := NewBusyboxPodWithName(E2eNamespace, "deletion-test")
-	duration := 1 * time.Minute
+	duration := assert.DefaultTimeout()
 	NewTestCase(t, e, "DeletePod", assert, "Deletion complete").WithPod(pod).WithDeleteAssertion(&duration).Run()
 }
 

--- a/test/e2e/ibmcloud_common.go
+++ b/test/e2e/ibmcloud_common.go
@@ -8,6 +8,7 @@ import (
 	"errors"
 	"strings"
 	"testing"
+	"time"
 
 	corev1 "k8s.io/api/core/v1"
 
@@ -285,6 +286,10 @@ func NewPodWithPVCFromIBMVPCBlockDriver(namespace, podName, containerName, image
 // IBMCloudAssert implements the CloudAssert interface for ibmcloud.
 type IBMCloudAssert struct {
 	VPC *vpcv1.VpcV1
+}
+
+func (c IBMCloudAssert) DefaultTimeout() time.Duration {
+	return 1 * time.Minute
 }
 
 func (c IBMCloudAssert) HasPodVM(t *testing.T, id string) {

--- a/test/e2e/libvirt_common.go
+++ b/test/e2e/libvirt_common.go
@@ -6,6 +6,7 @@ package e2e
 import (
 	"strings"
 	"testing"
+	"time"
 
 	"libvirt.org/go/libvirt"
 )
@@ -14,6 +15,10 @@ import (
 type LibvirtAssert struct {
 	// TODO: create the connection once on the initializer.
 	//conn libvirt.Connect
+}
+
+func (c LibvirtAssert) DefaultTimeout() time.Duration {
+	return 1 * time.Minute
 }
 
 func (l LibvirtAssert) HasPodVM(t *testing.T, id string) {

